bft_types/src/lib.rs

```diff
@@ -1,15 +1,16 @@
 //! Instruction types for the BF interpreter to use.
 
+use std::collections::HashMap;
 use std::fmt::Display;
 use std::fs;
-use std::path::{Path, PathBuf};
+use std::path::{Path, PathBuf};
 
-/// Types of Brainfuck instructions
-#[derive(Debug, PartialEq, Clone)]
-pub enum Instruction {
-    /// Increment the data pointer by one (to point to the next cell to the right).
+/// Types of Brainfuck instructions
+#[derive(Debug, PartialEq, Clone, Eq, Hash)]
```

Comment:
Adding `Hash` here seems worrying.  Personally the thing I think is missing from here is `Copy`